test/e2e/framework/create.go

```diff
@@ -623,3 +623,17 @@ func PrettyPrint(item interface{}) string {
 	}
 	return fmt.Sprintf("%+v", item)
 }
+
+// patchContainerImages replaces the specified Container Registry with a custom
+// one provided via the KUBE_TEST_REPO_LIST env variable
+func patchContainerImages(containers []v1.Container) error {
+	var err error
+	for _, c := range containers {
+		c.Image, err = imageutils.ReplaceRegistryInImageURL(c.Image)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
```

Comment:
I don't think this function should be part of the e2e test framework.
It think it is better suited in the e2epod (e2e/framework/pod) package it it deals with patching the image field of containers. I wouldn't expect a test framework to provide this functionality.
This was the initial reasoning for creating the e2epod package - to clean up the framework and cleanly separate functionality into different subpackages.

/hold

Reply:
Additional comments  https://github.com/kubernetes/kubernetes/pull/83814#discussion_r334217997

Reply:
The call to `patchContainerImges()` is made by [`framework.patchItemRecursively()`](https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/create.go#L310).
This method calls other functions to patch a name and a namespace.
These methods may be worth packing into a framework subpackage (i.e., `e2e/framework/patch`).

However, the method itself, `patchItemRecursively()` should stay within the framework as it is sued in https://github.com/kubernetes/kubernetes/blob/master/test/e2e/framework/create.go#L116

Reply:
I see your point, that makes sense to me.
I'd like to find another way to reduce circular dependencies between the core and subpackages.

/close